apps/explorer/src/components/ownedobjects/utils.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiMoveNormalizedType } from '@mysten/sui.js';
+
+type TypeReference =
+    | {
+          address: string;
+          module: string;
+          name: string;
+          typeArguments?: SuiMoveNormalizedType[];
+      }
+    | string
+    | number;
+
+export function extractSerializationType(
+    type: SuiMoveNormalizedType
+): TypeReference {
+    if (typeof type === 'string') {
+        return type;
+    }
+
+    if ('TypeParameter' in type) {
+        return type.TypeParameter;
+    }
+
+    if ('Reference' in type) {
+        return extractSerializationType(type.Reference);
+    }
+
+    if ('MutableReference' in type) {
+        return extractSerializationType(type.MutableReference);
+    }
+
+    if ('Vector' in type) {
+        return extractSerializationType(type.Vector);
+    }
+
+    if ('Struct' in type) {
+        const theType = type.Struct;
+        const theTypeArgs = theType.typeArguments;
+
+        if (theTypeArgs && theTypeArgs.length > 0) {
+            return extractSerializationType(theTypeArgs[0]);
```

Comment:
why do we do this? And why always the first element of the TypeArguments?

Reply:
Some types like SUI coins  have child sub type arguments, since we map this to a single key  we just need the first item